### PR TITLE
Removes duplicate alerts from table alerts

### PR DESCRIPTION
**Pull request by mmarchand**
Fix for duplicate alert entries

### CODE REVIEW

File: sql-schema/094.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+DELETE a1 FROM alerts a1, alerts a2 WHERE a1.id < a2.id AND a1.device_id = a2.device_id AND a1.rule_id = a2.rule_id;
+ALTER TABLE `alerts` ADD UNIQUE `unique_alert`(`device_id`, `rule_id`);
+
```
Comment (laf) [inline sql-schema/094.sql]: Can you remove the blank line :)

**Comment by laf** [comment]
Please rebase and use 098.sql

**Comment by f0o** [comment]
How's the testing going on this @mmarchand ?

Seen any duplicates since the SQL?

**Comment by mmarchand** [comment]
@f0o no more dups with this update, I believe you can merge it :)
thanks

**Comment by ekoyle** [comment]
Did we find what was creating the duplicates, or are we just going to let the unique constraint handle it?

**Comment by laf** [comment]
We've not had a proven way to replicate this so it's going to be hard to track down :/

**Comment by ekoyle** [comment]
Could it be at includes/polling/functions.inc.php around line 206?  It looks like there is an unconditional insert there into the alerts table... shouldn't that be eventlog instead alerts?

    if ($device['status'] != $status) {
        $poll_update   .= $poll_separator."`status` = '$status'";
        $poll_separator = ', ';

        dbUpdate(array('status' => $status, 'status_reason' => $response['status_reason']), 'devices', 'device_id=?', array($device['device_id']));
        dbInsert(array('importance' => '0', 'device_id' => $device['device_id'], 'message' => 'Device is '.($status == '1' ? 'up' : 'down')), 'alerts');

        log_event('Device status changed to '.($status == '1' ? 'Up' : 'Down'), $device, ($status == '1' ? 'up' : 'down'));
    }

**Comment by f0o** [comment]
Argh sorry big thumbs reverting

**Comment by f0o** [comment]
Terribly sorry for this mess.
@mmarchand can youbre submit the PR please?

@ekoyle the code you linked shouldnt exist anymore, its the old alerting that was replaced. Please open an issue for this a s reminder.

**Pull request merged**